src/Components/DrillAnimation.js

```diff
@@ -1,29 +1,245 @@
-import React, { Component } from 'react';
-import { StyleSheet, View, Text, Dimensions } from 'react-native';
-import Animation from './Animation';
+import React, { useState } from 'react';
+import { ScrollView, StyleSheet, View, Text, Dimensions, TouchableOpacity } from 'react-native';
 import { WebView } from 'react-native-webview';
+import { LinearGradient } from 'expo-linear-gradient';
+
+import Animation from './Animation';
+import VimeoVideo from './VimeoVideo';
+import theme from '../styles/theme.style';
+
+const DrillAnimation = props => {
+  const [count, setCount] = useState(0);
+
+  const incrementCount = () => {
+    setCount((count + 1) % (props.drill.steps.length + 1));
+  };
+
+  const displayNextStep = () => {
+    if (count + 1 === props.drill.steps.length) {
+      return (
+        <View style={styles.infoWrapper}>
+          <View style={styles.description}>
+            <View style={styles.wrapperFinish}>
+              <Text style={styles.fitnessNext}>Finish</Text>
+            </View>
+          </View>
+          <View style={styles.lines} />
+        </View>
+      );
+    } else {
+      return (
+        <View style={styles.infoWrapper}>
+          <View style={styles.description}>
+            <View style={styles.subWrapper}>
+              <Text style={styles.fitnessNext}>{props.drill.steps[count + 1].count}</Text>
+            </View>
+            <View style={styles.subSubWrapper}>
+              <Text style={styles.fitnessNext}>{props.drill.steps[count + 1].title}</Text>
+            </View>
+            <View style={styles.fakeWrapper} />
+          </View>
+          <View style={styles.lines} />
+        </View>
+      );
+    }
+  };
+
+  const checkSwitch = () => {
+    if (props.drill.steps && count === props.drill.steps.length) {
+      return displayFinish();
+    } else {
+      switch (props.drill.steps && props.drill.steps[count].source) {
+        case 'animation':
+          return displayAnimation();
+
+        case 'youtube':
+          return displayYoutube();
+
+        case 'vimeo':
+          return displayVimeo();
+
+        default:
+          return <Text>No visual content for this drill</Text>;
+      }
+    }
+  };
+
+  const displayFinish = () => {
+    return (
+      <View>
+        <View style={styles.containerFinish}>
+          <View style={styles.infoWrapper}>
+            <TouchableOpacity style={styles.buttonFinish} onPress={() => incrementCount()}>
```

Comment:
Maybe we should extract a button wince we use it on several pages and we are probably going to use it as well on the trainings

We already have a HeaderButton and a FilterButton. Is this one a `CtaButton`. I don't have a lot of ideas 😕

Reply:
I named it `GradientButton` ;)

Reply:
Much better than `CtaButton` 😄